manifests/server/etcd.pp

```diff
@@ -120,6 +121,17 @@
   }
 
   if $ensure == 'present' and $manage_members {
+    if defined(Class['k8s']) {
+      $_k8s_cluster_name = $k8s::etcd_cluster_name
+      $_k8s_puppetdb_discovery_tag = $k8s::puppetdb_discovery_tag
+    } else {
+      $_k8s_cluster_name = lookup('k8s::cluster_name', undef, undef, undef)
```

Comment:
If the user has in his manifest:

```puppet
class { 'k8s':
  cluster_name => 'acme',
}
```

This lookup() will *not* get 'acme', but whatever exist in hiera.

Same for most lookup() examples (I have not yet found a valid usage of this function in a module, and even in a control repo I consider it a smell).